### PR TITLE
Fixes #11855: Automatically package fusion dependencies

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/11855

### CODE REVIEW

**Comment by amousset** [comment]
This is great!

File: rudder-agent/SOURCES/mirror [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
```
Comment (amousset) [inline rudder-agent/SOURCES/mirror]: Could you document this script?

**Comment by peckpeck** [comment]
Commit modified

File: rudder-agent/SOURCES/mirror [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# Usage: ./mirror <destination_directory> [<Module::Name> ...]
+DIR="$1"
+shift
+
+# Use cpanm to download all dependencies from cpan
+cpanm --quiet -L /tmp/empty --self-contained --scandeps --save-dists "$DIR" "$@"
+
+# Download 3 index files mandatory for cpanm to work without connection to cpan.org
+mkdir -p "$DIR/authors"
+wget http://www.cpan.org/authors/01mailrc.txt.gz -O "$DIR/authors/01mailrc.txt.gz"
```
Comment (amousset) [inline rudder-agent/SOURCES/mirror]: https would be better

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**